app/models/mixins/retirement_mixin.rb

```diff
@@ -39,6 +39,10 @@ def set_retirement_requester(obj_ids, requester)
     end
   end
 
+  def make_retire_request(requester_id)
+    self.class.make_retire_request(id, User.find(requester_id), :initiated_by => 'user')
```

Comment:
eventually we can also keep initiated_by

```ruby
  def make_retire_request(requester_id, initiated_by: 'user')
     self.class.make_retire_request(id, User.find(requester_id), initiated_by: initiated_by)
  end
```

Reply:
We should pass along `initiated_by` if it is already set.

Reply:
No. We can but it seems pretty silly: the API method in 743 has to match this signature, and from the perspective of the API requester and initiator are both current user so it's pretty silly to have the api call be `queue_object_action(virtual_machine, desc, :method_name => "make_retire_request", :class_name => "VmOrTemplate", :role => "automate", :args => [User.current_user.id, User.current_user.id])`

Reply:
`:args => [User.current_user.id]` would work for `initiated_by: 'user'` - since that parameter would just get the default parameter.

Reply:
⬆️ That's exactly what I said in chat, Libor, but you said to make the method signature match as closely as I could.

Reply:
Right under the :bus: :laughing:

Reply:
> `:args => [User.current_user.id]` would work for `initiated_by: 'user'` - since that parameter would just get the default parameter.

@kbrock like this ?

```ruby
def make_retire_request(requester_id, initiated_by: 'user')
  self.class.make_retire_request(id, User.find(requester_id), initiated_by: initiated_by)
end
```